src/book.py

```diff
@@ -272,7 +272,10 @@ def snippet(self, length, on, groups=1, non_cont=False, randomized=True, rem_sto
                 
                 for index in sorted(range(start, start+length), reverse=True):
                     if (len(tokens)-length > 0):
-                        del tokens[index] 
+                        if 'char' in on.lower():
+                            tokens.replace(tokens[index],'')
```

Comment:
`str.replace` does not change the string, but returns a new string. You'd have to do
`tokens = tokens.replace(...)`.

At this point, though, I think I'm going to see if I can simplify this code at all.

Reply:
```
text = 'author'
text.replace(text[5],'')
```
returns `autho`. The way I am implementing the code needs exactly this.

Reply:
If it is char, tokens is a string object, otherwise tokens is an array. I think the behaviour is consistent now.

Reply:
Right, but if you do this:
```
text = 'author'
text.replace(text[5],'')
print(text)
```
You get
```
'author'
```

You need to do
```
text = text.replace(text[5], '')
```

Reply:
yes, my bad. I forgot the behaviour for strings. It should be ok now. Anyway, please feel free to simplify the code, but I definitely want an option to have with-replacement=False. If you think the behaviour is consistent, merge it so that others' codes do not break.

Reply:
Are you ever going to want `with_replacement=False` and `non_cont=False` at the same time?

Reply:
Yes. Suppose this:
Text = Sent1. Sent2. Sent3. Sent4. Sent5.
`length = 2, on='sent', groups=2, with-replacement=False, non-cont=False`  should give me
[Sent1, Sent2, Sent3, Sent4] or [Sent2, Sent3, Sent4, Sent5]

Reply:
Sorry I meant to ask if you'll ever want `with_replacement=True` and `non_cont=False` at the same time. In other words, as I understand it, if `non_cont=False`, once you've chosen a starting index, there's only one unique return given `groups` and `length`

I don't understand your example, though. Those are not length 2 snippets. Do you mean there are two possibilities:
[[Sent1, Sent2], [Sent3, Sent4]] or [[Sent2, Sent3], [Sent4, Sent5]]?

Reply:
Yes I mean that. But default behaviour is to join 2 snippets in a group to return a booktext object. That's why I wrote it that way.

Reply:
I will never need `with_replacement=True`.